scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/LazyTokenIterator.scala

```diff
@@ -63,17 +50,17 @@ private[parsers] class LazyTokenIterator private (
    * Deals with different rules for indentation after self type arrow.
    */
   def undoIndent(): Unit = {
-    sepRegions match {
+    curr.regions match {
       case (region: SepRegionIndented) :: others if curr.token.is[Indentation.Indent] =>
         next()
-        sepRegions = sepRegions match {
+        curr.regions = curr.regions match {
```

Comment:
```suggestion
        val reducedRegions = curr.regions match {
```
and then:
```
curr = curr.withRegions(newRegions)
```

we would need to keep regions a var in that case and have just `curr` to be mutable

Reply:
done

Reply:
@tgodzik one question about this change: i am planning to add another `var` to `TokenRef` (to make it into a linked list, for caching computations), and the whole point is not to make a copy when setting `.next` on it.

with that, do you still prefer `regions` to be a `val`?

Reply:
Or we could make it a method like:
```
curr.reduceRegions{
  case ...
}

```
so that we keep the var private, what do you think?

Reply:
sure, i can do that. although we don't really do any sort of vetting of the values assigned in `reduceRegions`, which makes it basically the same as a blind assignment. and `TokenRef` is a package-private class...

Reply:
i'll keep regions a `val` here.